File: api/src/Domain/Settings/Services/BusinessService.cs
using System;
using System.Threading.Tasks;
using DetailingArsenal.Domain.Users;

namespace DetailingArsenal.Domain.Settings {
    public interface IBusinessService : IService {
        Task<Business> GetById(Guid id);
        Task<Business> GetByUser(User user);
        Task<Business> CreateDefault(User user);
        Task Update(Business business, UpdateBusiness update);
    }

    public class BusinessService : IBusinessService {
        public IBusinessRepo repo;

        public BusinessService(IBusinessRepo repo) {
            this.repo = repo;
        }

        public async Task<Business> GetById(Guid id) {
            return await repo.FindById(id) ?? throw new EntityNotFoundException();
        }

        public async Task<Business> GetByUser(User user) {
            return await repo.FindByUser(user) ?? throw new EntityNotFoundException();
        }

        public async Task<Business> CreateDefault(User user) {
            var b = Business.Create(user.Id);
            await repo.Add(b);

            return b;
        }

        public async Task Update(Business business, UpdateBusiness update) {
            business.Name = update.Name;
            business.Address = update.Address;
            business.Phone = update.Phone;

            await repo.Update(business);
        }
    }
}

File: app/src/api/scheduling/settings/services/business-service.ts
import { http } from '@/api/core/http';
import { BusinessUpdate } from '@/api/scheduling/settings/data-transfer-objects/business-update';
import { Business } from '@/api/scheduling/settings/data-transfer-objects/business';

export class BusinessService {
    public async getBusiness() {
        var res = await http.get('/settings/business');

        const b = new Business(res.data.id, res.data.name, res.data.address, res.data.phone);

        return b;
    }

    public async updateBusiness(updateBusiness: BusinessUpdate) {
        var res = await http.put(`/settings/business/${updateBusiness.id}`, updateBusiness);

        const b = new Business(res.data.id, res.data.name, res.data.address, res.data.phone);

        return b;
    }
}
